internal/auth0/connection/schema.go

```diff
@@ -853,3 +855,92 @@ var optionsSchema = &schema.Schema{
+		Description: "ID of the connection for this SCIM configuration.",
+	},
+	"connection_name": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Name of the connection for this SCIM configuration.",
+	},
+	"strategy": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Schema of the connection for this SCIM configuration.",
+	},
+	"tenant_name": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Name of the tenant for this SCIM configuration.",
+	},
+	"user_id_attribute": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		RequiredWith: []string{"user_id_attribute", "mapping"},
+		Computed:     true,
+		Description:  "User ID attribute for generation unique of user ids. If `user_id_attribute` is set, `mapping` must be set as well. Defaults to `externalId`.",
+	},
+	"mapping": {
+		Type:         schema.TypeSet,
+		Optional:     true,
+		RequiredWith: []string{"user_id_attribute", "mapping"},
+		Computed:     true,
+		Description:  "Mapping between Auth0 attributes and SCIM attributes. If `mapping` is set, `user_id_attribute` must be set as well.",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"auth0": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+					Description:  "The field location in the Auth0 schema.",
+				},
+				"scim": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+					Description:  "The field location in the SCIM schema.",
+				},
+			},
+		},
+	},
+	"default_mapping": {
+		Type:        schema.TypeSet,
+		Computed:    true,
+		Description: "Default mapping between Auth0 attributes and SCIM attributes for this connectiion.",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"auth0": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The field location in the Auth0 schema.",
+				},
+				"scim": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The field location in the SCIM schema.",
+				},
+			},
+		},
+	},
+}
+
+var scimDataSourceSchema = getSCIMDataSourceSchema()
+
+func getSCIMDataSourceSchema() map[string]*schema.Schema {
+	dataSourceSchema := internalSchema.TransformResourceToDataSource(scimResourceSchema)
+	dataSourceSchema["connection_id"].Required = true
+	dataSourceSchema["connection_id"].Computed = false
+	dataSourceSchema["connection_id"].Optional = false
+	dataSourceSchema["user_id_attribute"].RequiredWith = []string{}
+	dataSourceSchema["user_id_attribute"].Description = "User ID attribute for generation unique of user ids."
+	dataSourceSchema["mapping"].RequiredWith = []string{}
+	dataSourceSchema["mapping"].Description = "Mapping between Auth0 attributes and SCIM attributes."
+	dataSourceSchema["mapping"].Elem = dataSourceSchema["default_mapping"].Elem
+
+	return dataSourceSchema
+}
```

Comment:
Let's follow the same pattern as for the other data sources and move this within the data source file. We should also not have this as a global private var within the package, as we can mistakenly modify it erroneously. 

Furthermore, let's move the `RequiredWith` within the `TransformResourceToDataSource` func, as it should have probably been there, given data sources won't have a need for this param to be set on read only properties.

Reply:
Actually, I don't think we need that at all, as we never copy that attribute in the first place. I am making a slight refactor here though to clean it up